vmware-base/src/com/cloud/hypervisor/vmware/util/VmwareHelper.java

```diff
@@ -79,6 +79,52 @@ public static boolean isReservedScsiDeviceNumber(int deviceNumber) {
         return deviceNumber == 7;
     }
 
+    public static VirtualDevice prepareNicOpaque(VirtualMachineMO vmMo, VirtualEthernetCardType deviceType, String portGroupName,
+            String macAddress, int deviceNumber, int contextNumber, boolean conntected, boolean connectOnStart) throws Exception {
+
+        assert(vmMo.getRunningHost().hasOpaqueNSXNetwork());
+
+        VirtualEthernetCard nic;
+        switch (deviceType) {
+        case E1000:
+            nic = new VirtualE1000();
+            break;
+
+        case PCNet32:
+            nic = new VirtualPCNet32();
+            break;
+
+        case Vmxnet2:
+            nic = new VirtualVmxnet2();
+            break;
+
+        case Vmxnet3:
+            nic = new VirtualVmxnet3();
+            break;
+
+        default:
+            assert (false);
```

Comment:
Why this "assert (false);" here? Asserting as false might cause problems.
```
Each assertion contains a boolean expression that you believe will be true when the assertion executes. If it is not true, the system will throw an error.
```

Source: http://docs.oracle.com/javase/7/docs/technotes/guides/language/assert.html

The user os assertion in this way is irrelevant and error-prone. Could you please remove it and push a new commit?

Cheers,
Wilder